extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClients.java

```diff
@@ -299,6 +269,39 @@ private MongoClientSettings createMongoConfiguration(MongoClientConfig config) {
         return settings.build();
     }
 
+    private void configureCodecRegistry(CodecRegistry defaultCodecRegistry, MongoClientSettings.Builder settings) {
+        List<CodecProvider> providers = new ArrayList<>();
+        if (!mongoClientSupport.getCodecProviders().isEmpty()) {
+            providers.addAll(getCodecProviders(mongoClientSupport.getCodecProviders()));
+        }
+        // add pojo codec provider with automatic capabilities
+        // it always needs to be the last codec provided
+        PojoCodecProvider.Builder pojoCodecProviderBuilder = PojoCodecProvider.builder()
+                .automatic(true)
+                .conventions(Conventions.DEFAULT_CONVENTIONS);
+        // register bson discriminators
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        for (String bsonDiscriminator : mongoClientSupport.getBsonDiscriminators()) {
+            try {
+                pojoCodecProviderBuilder
+                        .register(ClassModel.builder(Class.forName(bsonDiscriminator, true, classLoader))
+                                .enableDiscriminator(true).build());
+            } catch (ClassNotFoundException e) {
+                // Ignore
```

Comment:
it doesn't relate to the issue of the codec ordering, but ... suppressing of exception is not good- developers will spend a lot of time to understand why some logic doesn't work when it should be, it will be not so easy to find that something was broken because of class-load-issue... better will be to log it...

sorry for this my 2 cents

Reply:
@oxelad this error should never occurs as the class has been discover in a previous build step so it should be safe to ignore the error